Sprint 7 haunted house

jojo-ucf opened the pull request.

sherbracewell committed:
Enemies added

 - Flying Enemy added
- Flashlight has "cone of light"
- JumpScare ghost added
- HH Enemy Added

TriOmega committed:
Added basic enemy stunning

- Added IStunnable interface which helps enforce the mechanisms needed for stunning on enemies to work (but not all such as the coroutine)
- Modified standard and medium enemy scripts to disable animator and nav mesh agent when hit by flashlight
- Commented out line of code on SpecialHHEnemy that was causing a warning due to unused animator parameter

TriOmega committed:
Fix flashlight jitter

- Fixed flashlight jitter by refactoring portions of code between Update() and FixedUpdate()
- Tried a bunch of different stuff in order to try to get flashlight to stop detecting CheckpointTriggerVolumes and to get only the front 180 degrees range of motion in front of the player
- Updated Player prefab by slightly moving flashlight away from player

TriOmega committed:
Add temporary stun indicator

- Added prefab for temporary stun indicator object
- Added code to Standard and Medium enemies that instantiates a stun indication to give visual feedback for getting stunned

TriOmega reviewed (commented):
Needed Corrections For Next Time:
Remember to add _01 suffix to appropriate files
Scene file names shouldn't have underscores or spaces (this correction won't be made today in fear that it will mess with the sprint building process)

Notes:
CharController still has a Debug.Log on  line 120

TriOmega committed:
Update animation controller file name

- AC_FlyingEnemy_01

TriOmega reviewed (commented):
Notes:
Flashlight is going to look weird because the material used for it is from particles and was either changed by Gabe or is displayed differently with the new render pipeline, change suggested
In the future as well try not to use a material for something that it wasn't intended specifically for

TriOmega committed:
Merge branch 'main' into sprint-7-haunted-house

TriOmega committed:
Update JodyScene Skybox & InputManager

- Returned skybox material in JodyScene to default as not to cause after-image graphical glitch in the new render pipeline
- InputManager was updated due to the render pipeline with debug controls
